processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -291,6 +296,38 @@ public int lookupId(String name)
           }
         };
       }
+
+      @Override
+      public NumericColumnSelector makeMathExpressionSelector(String expression)
+      {
+        final Expr parsed = Parser.parse(expression);
+
+        final Set<String> required = Sets.newHashSet(Parser.findRequiredBindings(parsed));
+        final Map<String, Supplier<Number>> values = Maps.newHashMapWithExpectedSize(required.size());
```

Comment:
these two are repeated everywhere, is it appropriate to change method signature to take arguments (Expression  expr, List<String> requiredBindings) ?